### PR TITLE
Add headless notices at end of cmd exec

### PR DESCRIPTION
**Pull request by daved**
https://www.pivotaltracker.com/story/show/175020204

### CODE REVIEW

File: cmd/state/internal/headless/headless.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+func Notify(prime *primer.Values, err error, cmdNames ...string) {
+	if err != nil || !prime.Project().IsHeadless() {
+		return
+	}
+
+	for _, cmd := range cmdNames {
+		prime.Output().Error(locale.T("message_headless_" + cmd))
+	}
+
+	prime.Output().Error(locale.T("message_headless"))
```
Comment (Naatan) [inline cmd/state/internal/headless/headless.go:21]: Use `Notice()` not `Error()`

**Review by Naatan — CHANGES_REQUESTED** [review]
I'd rather see this in the runner. cmdtree should just be dealing with the command registration and routing, it should not be printing anything.

Other than that this looks ok for a first iteration. Not sure about the headless package, but I can't think of anything better off the top of my head.

File: cmd/state/internal/headless/headless.go (hunk 1 above)
Comment (daved) [reply cmd/state/internal/headless/headless.go:21]: :facepalm: I didn't even think about whether notice was already going to stderr.

File: internal/runners/packages/import.go [hunk 2]
```diff
@@ -48,25 +49,34 @@ func NewImportRunParams() *ImportRunParams {
...
 		prime.Output(),
+		prime.Project(),
 		prime.Prompt(),
 	}
 }
 
 // Run executes the import behavior.
 func (i *Import) Run(params ImportRunParams) error {
+	err := i.run(params)
+	headless.Notify(i.out, i.proj, err)
```
Comment (Naatan) [inline internal/runners/packages/import.go:75]: missing cmdNames

File: internal/runners/packages/search.go [hunk 3]
```diff
@@ -19,18 +19,26 @@ type SearchRunParams struct {
...
-		out: prime.Output(),
+		out:  prime.Output(),
+		proj: prime.Project(),
 	}
 }
 
 // Run is executed when `state packages search` is ran
 func (s *Search) Run(params SearchRunParams) error {
+	err := s.run(params)
+	headless.Notify(s.out, s.proj, err)
```
Comment (Naatan) [inline internal/runners/packages/search.go:37]: Missing cmdNames

File: internal/runners/packages/list.go [hunk 4]
```diff
@@ -23,18 +23,26 @@ type ListRunParams struct {
...
-		out: prime.Output(),
+		out:  prime.Output(),
+		proj: prime.Project(),
 	}
 }
 
 // Run executes the list behavior.
 func (l *List) Run(params ListRunParams) error {
+	err := l.run(params)
+	headless.Notify(l.out, l.proj, err)
```
Comment (Naatan) [inline internal/runners/packages/list.go:41]: missing cmdNames

**Review by Naatan — CHANGES_REQUESTED** [review]

File: internal/runners/packages/import.go (hunk 2 above)
Comment (daved) [reply internal/runners/packages/import.go:75]: Theses are supposed to be missing the cmdNames; They should only print a general message, and not a command-specific message.

Comment (Naatan) [reply internal/runners/packages/import.go:75]: Where do you get that from? Looking at the AC's again I don't see these commands in there at all.

Comment (daved) [reply internal/runners/packages/import.go:75]: hmm. I might have misunderstood. Pete had stated that all commands run headless should get a generic message, but some should get a more specific message. I'll review the AC and what Pete said in Slack and update the AC, then ping you for approval.

Comment (Naatan) [reply internal/runners/packages/import.go]: Please do not update ACs without my say so. Slack does not trump ACs for stories that are in our sprint commit.

**Review by Naatan — APPROVED** [review]

**Pull request merged**